### PR TITLE
feat: `createRequire` option and disable esm for jest

### PR DESCRIPTION
**Pull request by clarkdo**
## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### esm disabled by default

As [esm](https://github.com/standard-things/esm) package has been incompatible since [jest 25](https://jestjs.io/blog/2020/01/21/jest-25), this PR disables esm for jest environments by default to prevent regression happening when testing nuxt with jest. You may need to make changes to your project after this PR:

jest config:

```js
    "transform": {
      "^.+\\.js$": "babel-jest"
    },
```

babel config:

```js
module.exports = function (api) {
  if (api.env('test')) {
    return {
      plugins: ['@babel/plugin-transform-modules-commonjs']
    }
  }
  return {}
}
```

### `createRequire` option

Meanwhile, we have started working on [`jiti`](https://github.com/nuxt-contrib/jiti) project for stable esm/typescript runtime support. You can try it by setting `createRequire: 'jiti'` inside `nuxt.config`.

Also, it is possible to use this option to keep using `esm` with jest < 25 or provide a custom factory function. Setting value to false or unsupported value, makes nuxt using native require (via [create-require](https://github.com/nuxt-contrib/create-require))

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

**Comment by clarkdo** [comment]
@pi0 Feel free to merge or close if we're gonna remove esm.

**Comment by pi0** [comment]
@clarkdo PR is good we should remove esm more carefully. I was just thinking to provide an option instead of disabling. Can make changes in same PR :)

File: packages/config/src/load.js [hunk 1]
```diff
@@ -52,9 +53,8 @@ export async function loadNuxtConfig ({
   if (configFile) {
     // Clear cache
     clearRequireCache(configFile)
-
-    options = esm(module)(configFile) || {}
-
+    const requirModule = createRequire(module)
```
Comment (Atinux) [inline packages/config/src/load.js:56]:
```suggestion
    const requireModule = createRequire(module)
```

File: packages/config/src/load.js [hunk 2]
```diff
@@ -52,9 +53,8 @@ export async function loadNuxtConfig ({
...
+    options = requirModule(configFile) || {}
```
Comment (Atinux) [inline packages/config/src/load.js:57]:
```suggestion
    options = requireModule(configFile) || {}
```

File: packages/config/src/load.js (hunk 1 above)
Comment (pi0) [reply packages/config/src/load.js:56]: @Atinux PR is still WIP pushing fixes in few minutes :)

**Pull request merged**